workflows/phylogenetics/wf_snippy_tree.wdl

```diff
@@ -36,36 +38,41 @@ workflow snippy_tree_wf {
         alignment = snippy_core.snippy_full_alignment_clean,
         cluster_name = tree_name
     }
-    call snp_dists.snp_dists as snp_dists_gubbins {
-      input:
-        alignment = gubbins.gubbins_polymorphic_fasta,
-        cluster_name = tree_name
-    }
-    call reorder_matrix.reorder_matrix as reorder_matrix_gubbins {
-      input:
-        input_tree = gubbins.gubbins_final_labelled_tree,
-        matrix = snp_dists_gubbins.snp_matrix,
-        cluster_name = tree_name 
+    if (core_genome) {
+      call snp_sites_task.snp_sites as snp_sites_gubbins{
+        input:
+          msa_fasta = gubbins.gubbins_polymorphic_fasta,
+          output_name = tree_name,
+          output_multifasta = true
+      }
     }
   }
   if (!use_gubbins) {
-    call iqtree_task.iqtree {
-      input:
-        alignment = snippy_core.snippy_full_alignment_clean,
-        cluster_name = tree_name
-    }
-    call snp_dists.snp_dists as snp_dists_iqtree {
-      input:
-        alignment = snippy_core.snippy_full_alignment_clean,
-        cluster_name = tree_name
-    }
-    call reorder_matrix.reorder_matrix as reorder_matrix_iqtree {
-      input:
-        input_tree = iqtree.ml_tree,
-        matrix = snp_dists_iqtree.snp_matrix,
-        cluster_name = tree_name 
+    if (core_genome) {
+      call snp_sites_task.snp_sites as snp_sites_no_gubbins {
+        input:
+          msa_fasta = snippy_core.snippy_full_alignment_clean,
+          output_name = tree_name,
+          output_multifasta = true
+      }
     }
   }
+  call iqtree_task.iqtree {
+    input:
+      alignment = select_first([snp_sites_gubbins.snp_sites_multifasta, gubbins.gubbins_polymorphic_fasta, snp_sites_no_gubbins.snp_sites_multifasta, snippy_core.snippy_full_alignment_clean]),
+      cluster_name = tree_name
+  }
+  call snp_dists.snp_dists as snp_dists_iqtree {
```

Comment:
You can probably remove the _iqtree suffix here since we're no longer doing a separate _gubbins vs _iqtree version